test/integration/executor/create_asset_test.cpp

```diff
@@ -0,0 +1,114 @@
+
+static const AssetNameType kAssetName{"new_asset"};
+constexpr PrecisionType kAssetPrecision(1);
+
+const AssetIdType &getNewId() {
+  static const AssetIdType kNewId{kAssetName + "#" + kDomain};
+  return kAssetId;
+}
+
+class CreateAssetTest : public ExecutorTestBase {
+ public:
+
+  void checkNoSuchAsset(
+      const boost::optional<AssetIdType> asset_id = boost::none) {
+    auto asset_val = asset_id.value_or(getNewId());
+    checkQueryError<shared_model::interface::NoAssetErrorResponse>(
+        getItf().executeQuery(
+            *getItf().getMockQueryFactory()->constructCreateAsset(
+                asset_val)),
+        0);
+  }
+
+  iroha::ametsuchi::CommandResult createAsset(
+      const AccountIdType &issuer,
+      const AssetNameType &target_name = kAssetName,
+      const DomainIdType &target_domain = kDomain,
+      constexpr PrecisionType  &precision = kAssetPrecision,
+      bool validation_enabled = true) {
+    return getItf().executeCommandAsAccount(
+        *getItf().getMockCommandFactory()->constructCreateAsset(
+            target_name, target_domain, precision),
+        issuer,
+        validation_enabled);
+  }
+//
+  iroha::ametsuchi::CommandResult createDefaultAsset(
+      const AccountIdType &issuer, bool validation_enabled = true) {
+    return createAsset(
+        issuer, kAssetName, kDomain, kAssetPrecision, validation_enabled);
+  }
+};
+
+using CreateAssetBasicTest = BasicExecutorTest<CreateAssetTest>;
+
+/**
+ *  given a user with all related permissions
+ *  when executes CreateAsset command with nonexistent domain
+ *  then the command does not succeed and the asset is not added
+ */
+TEST_P(CreateAssetBasicTest, NoDomain) {
+  checkCommandError(checkNoSuchAsset(kAdminId, kAssetName, "no_such_domain"), 3);
+  checkNoSuchAsset(kAssetName + "@no_such_domain");
+}
+
+/**
+ * given a user with all related permissions
+ * when executes CreateAsset command with already taken name
+ * then the command does not succeed and the original asset is not changed
+ */
+TEST_P(CreateAssetBasicTest, NameExists) {
+  ASSERT_NO_FATAL_FAILURE(
+      getItf().createAssetWithPerms(kAssetName, kDomain, kAssetPrecision, {}));
+  ASSERT_NO_FATAL_FAILURE(checkVerifiedProposal( 
+  			[](auto &vproposal) {
+            ASSERT_EQ(vproposal->transactions().size(), 0););
+
+  checkCommandError(createDefaultAsset(kAdminId), 4);
+  checkNoSuchAsset();
+}
+
+
+INSTANTIATE_TEST_SUITE_P(Base,
+                         CreateAssetBasicTest,
+                         executor_testing::getExecutorTestParams(),
+                         executor_testing::paramToString);
+
+using CreateAssetPermissionTest =
+    command_permission_test::CommandPermissionTest<CreateAssetTest>;
+
+TEST_P(CreateAssetPermissionTest, CommandPermissionTest) {
+  ASSERT_NO_FATAL_FAILURE(getItf().createDomain(kDomain));
+
+  if (checkResponse(createDefaultAsset(getActor(), getValidationEnabled()))) {
+    checkNoSuchAsset();
+  }

```

Comment:
```suggestion
  if (checkResponse(createDefaultAsset(getActor(), getValidationEnabled()))) {
    checkAsset();
  } else {
    checkNoSuchAsset();
  }

```
checkResponse returns `true` if the command succeeded. and if it succeeded we have to check that asset was created. otherwise that it was not.